//! @file ArmDebugger/UI/EditDisplayOptionsDialog.hpp
//! @brief The declaration of an object representing a dialog used to edit
//! the display of emulated memory.
//! @author GiantRobotLemur@na-se.co.uk
//! @date 2024
//! @copyright This file is part of the Mighty Oak project which is released
//! under LGPL 3 license. See LICENSE file at the repository root or go to
//! https://github.com/GiantRobotLemur/MightyOak for full license details.
////////////////////////////////////////////////////////////////////////////////

#ifndef __ARM_DEBUGGER_EDIT_DISPLAY_OPTIONS_DIALOG_HPP__
#define __ARM_DEBUGGER_EDIT_DISPLAY_OPTIONS_DIALOG_HPP__

////////////////////////////////////////////////////////////////////////////////
// Dependent Header Files
////////////////////////////////////////////////////////////////////////////////
#include <QDialog>

#include "ui_EditDisplayOptionsDialog.h"
#include "MemoryViewOptions.hpp"

////////////////////////////////////////////////////////////////////////////////
// Macro Definitions
////////////////////////////////////////////////////////////////////////////////

namespace Mo {

////////////////////////////////////////////////////////////////////////////////
// Data Type Declarations
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
// Class Declarations
////////////////////////////////////////////////////////////////////////////////
//! @brief An object representing a dialog used to edit the display of
//! emulated memory.
class EditDisplayOptionsDialog : public QDialog
{
public:
    // Construction/Destruction
    EditDisplayOptionsDialog(QWidget *ownerWidget);
    virtual ~EditDisplayOptionsDialog() = default;

    // Accessors
    const MemoryViewOptions &getOptions() const;
    void setOptions(const MemoryViewOptions &options);

    // Operations

private:
    // Internal Types

    // Internal Functions
    void updateToUI();
    void updateFromUI();

    // Internal Fields
    Ui::EditDisplayOptionsDialog _ui;
    MemoryViewOptions _options;
};

////////////////////////////////////////////////////////////////////////////////
// Function Declarations
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
// Templates
////////////////////////////////////////////////////////////////////////////////

} // namespace Mo

#endif // Header guard
////////////////////////////////////////////////////////////////////////////////
